src/System.Diagnostics.Process/src/System/Diagnostics/Process.Win32.cs

```diff
@@ -21,6 +23,11 @@ public partial class Process : IDisposable
 
         private bool StartCore(ProcessStartInfo startInfo)
         {
+            if (startInfo != null && startInfo.ArgumentList.Count > 0)
+            {
+                startInfo.Arguments = PasteArguments.Paste(startInfo.ArgumentList, true);
```

Comment:
Same comment as on Unix; I don't think we should be mutating the ProcessStartInfo object.

Reply:
As with the other comment, `false` is the correct value to pass here.

Reply:
On Unix, it doesn't make sense to spend cycles computing a flattened argument list if one isn't provided. At the end, Unix wants the arg array. The ArgumentList provided should be flowed down as an array (you'll have to preprend the argv[0] part, I assume.)